sdks/go/pkg/beam/core/runtime/harness/harness.go

```diff
@@ -98,6 +111,18 @@ func Main(ctx context.Context, loggingEndpoint, controlEndpoint string) error {
 		log.Debugf(ctx, "control response channel closed")
 	}()
 
+	// if the runner supports worker status api then expose SDK harness status
+	if statusEndpoint != "" {
+		statusHandler, err := newWorkerStatusHandler(ctx, statusEndpoint)
+		if err != nil {
+			log.Error(ctx, err)
```

Comment:
If we return an error response, is it worth trying to handle requests or should we stick the remaining logic in an else? I'd imagine that its just going to fail since we failed to set up our connection.

Reply:
yes, it just logs that it is not able to make a connection and fails quietly as this is optional feature for pipeline execution.